src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,247 @@
+// eslint-disable-next-line
+import { useParams } from 'react-router';
+import { useEffect, useState } from 'react';
+import { Link, useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
 
-export const PeoplePage = () => {
-  return (
-    <>
-      <h1 className="title">People Page</h1>
+export const PeoplePage: React.FC<{
+  isError: boolean;
+  setIsError: React.Dispatch<React.SetStateAction<boolean>>;
+  isLoading: boolean;
+  peopleFromServer: Person[] | undefined;
+  errorMessage: string | undefined;
+}> = ({
+  isError, setIsError, isLoading, peopleFromServer, errorMessage,
+}) => {
+  const { slug = '' } = useParams();
+  const [visiblePeople, setVisiblePeople] = useState(peopleFromServer || []);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const initialSort = searchParams.get('sort');
+  const [sortUpdate, SetSortUpdate] = useState(initialSort);
+  const [sortBefore, SetSortBefore] = useState(sortUpdate);
```

Comment:
```suggestion
  const [sortUpdate, setSortUpdate] = useState(initialSort);
  const [sortBefore, setSortBefore] = useState(sortUpdate);
```
fix all similar cases

Reply:
Also, I guess you can get rid of one of this state

Reply:
Unfortunately no. 
"Initial" is completely updated on rerendering. 
"Update" only when there is a request. 
"Before" his past state.